### PR TITLE
Use new platform class in cmd client

### LINKED ISSUE #10185: owncloudcmd fails with `not existant or bad metadata "owncloudsync_vfs_off"` error
### Pre-submission Checks

- [X] I [checked for similar issues](https://github.com/owncloud/client/issues?q=), but could not find any. I also checked the closed issues. I could not contribute additional information to any existing issue.
- [X] I will take the time to fill in all the required fields. I know that the bug report may be dismissed otherwise due to lack of information.

### Describe the QA issue

While running `owncloudcmd` to sync files, the command fails with following error:
```bash
[ critical plugins ]:	Could not load plugin: not existant or bad metadata "owncloudsync_vfs_off"
```

### Steps to reproduce the issue

1. Install `owncloud-client` from https://download.owncloud.com/desktop/ownCloud/daily/3.0/linux/Ubuntu_20.04/
2. Run the command:
```bash
/usr/bin/owncloudcmd --trust /home/demo/owncloud/ http://admin:admin@localhost/core
```

Maybe some libraries are missing out in https://download.owncloud.com/desktop/ownCloud/daily/3.0/linux/Ubuntu_20.04/

#### Information
OS: `Ubuntu 20.04`
Client: `3.0`

**NOTE: same for `Ubuntu_22.04`**

### Expected behavior

To complete sync

### Actual behavior

Fails with error

### PR DESCRIPTION
**Pull request by fmoc**
Fixes #10185.

### CODE REVIEW

File: src/cmd/cmd.cpp [hunk 1]
```diff
@@ -416,13 +418,13 @@ CmdOptions parseOptions(const QStringList &app_args)
 
 int main(int argc, char **argv)
 {
+    auto platform = OCC::Platform::create();
+
+    platform->migrate();
```
Comment (TheOneRing) [inline src/cmd/cmd.cpp:423]: Mirgrate reqires QApplication

File: src/cmd/cmd.cpp [hunk 2]
```diff
@@ -14,17 +14,18 @@
...
 #include <QFile>
 #include <QFileInfo>
 #include <QJsonDocument>
 #include <QJsonObject>
 #include <QNetworkProxy>
+#include <QStringList>
+#include <QUrl>
+#include <iostream>
+#include <memory>
+#include <qcoreapplication.h>
```
Comment (TheOneRing) [inline src/cmd/cmd.cpp:26]: Please use QCoreApplication

File: src/cmd/cmd.cpp [hunk 3]
```diff
@@ -14,17 +14,18 @@
...
-#include <QUrl>
 #include <QFile>
 #include <QFileInfo>
 #include <QJsonDocument>
 #include <QJsonObject>
 #include <QNetworkProxy>
+#include <QStringList>
+#include <QUrl>
+#include <iostream>
+#include <memory>
```
Comment (TheOneRing) [inline src/cmd/cmd.cpp:25]: Plesase seperate std and qt includes

**Review by TheOneRing — CHANGES_REQUESTED** [review]

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests</strong> https://drone.owncloud.com/owncloud/client/13574/6/1

:boom: The GUI tests failed.

GUI Logs: https://cache.owncloud.com/public/owncloud/client/13574/guiReportUpload/index.html

<details>

Server Logs: https://cache.owncloud.com/public/owncloud/client/13574/guiReportUpload/serverlog.log

Screenshots:
- [sharee_creates_a_file_and_a_folder_inside_a_shared_folder.png](https://cache.owncloud.com/public/owncloud/client/13574/guiReportUpload/screenshots/sharee_creates_a_file_and_a_folder_inside_a_shared_folder.png)

</details>

File: src/libsync/platform_mac.mm [hunk 4]
```diff
@@ -29,53 +37,45 @@ @implementation OwnAppDelegate {
...
 
 MacPlatform::MacPlatform()
 {
+    Q_D(MacPlatform);
+
     NSApplicationLoad();
-    _appDelegate = [[OwnAppDelegate alloc] init];
-    [[NSApplication sharedApplication] setDelegate:_appDelegate];
+    d->appDelegate = [[OwnAppDelegate alloc] init];
+    [[NSApplication sharedApplication] setDelegate:d->appDelegate];
```
Comment (fmoc) [inline src/libsync/platform_mac.mm:65]: This doesn't work, I'm not sure why. @erikjv can you help?

Comment (fmoc) [reply src/libsync/platform_mac.mm:65]: It segfaults due to an allocation error. Please try to run either `owncloud` or `owncloudcmd`.

Comment (TheOneRing) [reply src/libsync/platform_mac.mm:64]: I assume it is using the wrong objective c syntax to access the pointer?

Comment (erikjv) [reply src/libsync/platform_mac.mm:65]: The `d_ptr` is never initialised. Do something like:
```
MacPlatform::MacPlatform()
    : d_ptr(new MacPlatformPrivate)
{
```

Comment (erikjv) [reply src/libsync/platform_mac.mm:65]: Btw the ObjC bits are fine.

**Pull request merged**